src/libraries/System.Text.Json/tests/Serialization/CollectionTests/CollectionTests.Dictionary.NonStringKey.cs

```diff
@@ -574,5 +575,49 @@ public class SuffixNamingPolicy : JsonNamingPolicy
             public const string Suffix = "_Suffix";
             public override string ConvertName(string name) => name + Suffix;
         }
+
+        [Fact]
+        public static void RoundtripAllDictionaryConverters()
+        {
+            const string Expected = @"{""1"":1}";
+
+            foreach (Type type in CollectionTestTypes.DeserializableDictionaryTypes<int, int>())
```

Comment:
Could this be refactored using Theory/MemberData for better test reporting?

Reply:
Yes we could do that. Can tackle in master/6.0 to keep this PR as a direct port.